modules/core/src/main/java/org/apache/ignite/internal/util/nio/GridNioServer.java

```diff
@@ -2878,6 +2862,19 @@ protected boolean close(final GridSelectorNioSessionImpl ses, @Nullable final Ig
                         req.onError(err);
                 }
 
+                try {
+                    filterChain.onSessionClosed(ses);
+                }
+                catch (IgniteCheckedException e1) {
+                    filterChain.onExceptionCaught(ses, e1);
+                }
+
+                if (outRecovery != null)
```

Comment:
As I can see the old version of code releases in and out recovery descriptors in finally block around cleaning up message requests.

In new version of code no finally block is protecting us from any exception there.

Do you see any risks here?